tests/agent/test_event_handler.py

```diff
@@ -193,3 +201,37 @@ def test__get_major_version_raises_on_invalid_version(self):
                     message="Creating a LegacyVersion has been deprecated and will be removed in the next major release",
                 )
                 _get_major_version("invalid_version")
+
+
+def test_parse_event_extra_field(example_event):
+    event_dict = example_event.dict()
+    event_dict["new_field"] = "surprise!"
+    serialized_bytes = json.dumps(dict(event_dict), indent=2).encode("utf-8")
+    event = EventHandler.parse_event_from(serialized_bytes)
+
+    assert event.type == "unknown_event"
+
+
+def test_parse_event_missing_required_field(example_event):
+    event_dict = example_event.dict()
+    del event_dict["datasource_name"]
```

Comment:
```suggestion
    event_dict = example_event.dict(exclude={"datasource_name"})
```

Reply:
oh this is smooth

Reply:
impl [here](https://github.com/great-expectations/cloud/pull/164/commits/481bbf0d4dbe4163aff47702a5a2b68cfcadeed6) in this PR: https://github.com/great-expectations/cloud/pull/164/commits/481bbf0d4dbe4163aff47702a5a2b68cfcadeed6